src/plugins/vis_type_timeseries/public/application/components/vis_editor.tsx

```diff
@@ -182,6 +195,48 @@ export class VisEditor extends Component<TimeseriesEditorProps, TimeseriesEditor
       >
         <DefaultIndexPatternContext.Provider value={this.state.defaultIndex}>
           <div className="tvbEditor" data-test-subj="tvbVisEditor">
+            {!hideCallout && (
+              <EuiCallOut
```

Comment:
nit: maybe we can move it into a new component `UseIndexPatternModeCallout` and use it like:  
`{ useKibanaIndexes && <UseIndexPatternModeCallout /> }`

then we can use localStorage in more React way e.g. using custom hook for that or use existing  https://github.com/streamich/react-use/blob/master/docs/useLocalStorage.md

It allows us to don't execute the code below after setting new value into `localStorage`
```
this.setState({
      hideCallout: true,
    }); 
```

Reply:
Solved by @alexwizp ❤️